File: src/gen1/precomp/instructions.pl
$VAR1 = {
          'any' => {
                     'instr' => 996,
                     'mnem' => 'any',
                     'opcode' => '000003e4',
                     'size' => 4
                   },
          'backcommit' => {
                            'instr' => 263104,
                            'mnem' => 'backcommit',
                            'opcode' => '000403c0',
                            'param1' => 'address',
                            'size' => 8,
                            'terse' => 1
                          },
          'call' => {
                      'instr' => 263042,
                      'mnem' => 'call',
                      'opcode' => '00040382',
                      'param1' => 'address',
                      'size' => 8,
                      'terse' => 1
                    },
          'catch' => {
                       'instr' => 263059,
                       'mnem' => 'catch',
                       'opcode' => '00040393',
                       'param1' => 'address',
                       'size' => 8,
                       'terse' => 1
                     },
          'char' => {
                      'instr' => 263127,
                      'mnem' => 'char',
                      'opcode' => '000403d7',
                      'param1' => 'char',
                      'size' => 8
                    },
          'closecapture' => {
                              'instr' => 262912,
                              'mnem' => 'closecapture',
                              'opcode' => '00040300',
                              'param1' => 'slot',
                              'size' => 8,
                              'terse' => 1
                            },
          'commit' => {
                        'instr' => 262966,
                        'mnem' => 'commit',
                        'opcode' => '00040336',
                        'param1' => 'address',
                        'size' => 8,
                        'terse' => 1
                      },
          'condjump' => {
                          'instr' => 525089,
                          'mnem' => 'condjump',
                          'opcode' => '00080321',
                          'param1' => 'register',
                          'param2' => 'address',
                          'size' => 12
                        },
          'counter' => {
                         'instr' => 525142,
                         'mnem' => 'counter',
                         'opcode' => '00080356',
                         'param1' => 'register',
                         'param2' => 'value',
                         'size' => 12
                       },
          'end' => {
                     'instr' => 262360,
                     'mnem' => 'end',
                     'opcode' => '000400d8',
                     'param1' => 'code',
                     'size' => 8,
                     'terse' => 1
                   },
          'endreplace' => {
                            'instr' => 921,
                            'mnem' => 'endreplace',
                            'opcode' => '00000399',
                            'size' => 4,
                            'terse' => 1
                          },
          'fail' => {
                      'instr' => 843,
                      'mnem' => 'fail',
                      'opcode' => '0000034b',
                      'size' => 4,
                      'terse' => 1
                    },
          'failtwice' => {
                           'instr' => 912,
                           'mnem' => 'failtwice',
                           'opcode' => '00000390',
                           'size' => 4
                         },
          'jump' => {
                      'instr' => 262963,
                      'mnem' => 'jump',
                      'opcode' => '00040333',
                      'param1' => 'address',
                      'size' => 8,
                      'terse' => 1
                    },
          'maskedchar' => {
                            'instr' => 525157,
                            'mnem' => 'maskedchar',
                            'opcode' => '00080365',
                            'param1' => 'char',
                            'param2' => 'mask',
                            'size' => 12,
                            'terse' => 1
                          },
          'noop' => {
                      'instr' => 0,
                      'mnem' => 'noop',
                      'opcode' => '00000000',
                      'size' => 4
                    },
          'opencapture' => {
                             'instr' => 263068,
                             'mnem' => 'opencapture',
                             'opcode' => '0004039c',
                             'param1' => 'slot',
                             'size' => 8,
                             'terse' => 1
                           },
          'partialcommit' => {
                               'instr' => 263092,
                               'mnem' => 'partialcommit',
                               'opcode' => '000403b4',
                               'param1' => 'address',
                               'size' => 8
                             },
          'quad' => {
                      'instr' => 263038,
                      'mnem' => 'quad',
                      'opcode' => '0004037e',
                      'param1' => 'quad',
                      'size' => 8
                    },
          'range' => {
                       'instr' => 525245,
                       'mnem' => 'range',
                       'opcode' => '000803bd',
                       'param1' => 'from',
                       'param2' => 'until',
                       'size' => 12,
                       'terse' => 1
                     },
          'replace' => {
                         'instr' => 525128,
                         'mnem' => 'replace',
                         'opcode' => '00080348',
                         'param1' => 'slot',
                         'param2' => 'address',
                         'size' => 12,
                         'terse' => 1
                       },
          'ret' => {
                     'instr' => 928,
                     'mnem' => 'ret',
                     'opcode' => '000003a0',
                     'size' => 4,
                     'terse' => 1
                   },
          'set' => {
                     'instr' => 2098122,
                     'mnem' => 'set',
                     'opcode' => '002003ca',
                     'param1' => 'set',
                     'size' => 36,
                     'terse' => 1
                   },
          'skip' => {
                      'instr' => 262960,
                      'mnem' => 'skip',
                      'opcode' => '00040330',
                      'param1' => 'number',
                      'size' => 8
                    },
          'span' => {
                      'instr' => 2098145,
                      'mnem' => 'span',
                      'opcode' => '002003e1',
                      'param1' => 'set',
                      'size' => 36
                    },
          'testany' => {
                         'instr' => 262918,
                         'mnem' => 'testany',
                         'opcode' => '00040306',
                         'param1' => 'address',
                         'size' => 8
                       },
          'testchar' => {
                          'instr' => 525210,
                          'mnem' => 'testchar',
                          'opcode' => '0008039a',
                          'param1' => 'address',
                          'param2' => 'char',
                          'size' => 12
                        },
          'testquad' => {
                          'instr' => 525275,
                          'mnem' => 'testquad',
                          'opcode' => '000803db',
                          'param1' => 'address',
                          'param2' => 'quad',
                          'size' => 12
                        },
          'testset' => {
                         'instr' => 2360163,
                         'mnem' => 'testset',
                         'opcode' => '00240363',
                         'param1' => 'address',
                         'param2' => 'set',
                         'size' => 40
                       },
          'trap' => {
                      'instr' => 4278255615,
                      'mnem' => 'trap',
                      'opcode' => 'ff00ffff',
                      'size' => 4
                    },
          'var' => {
                     'instr' => 263150,
                     'mnem' => 'var',
                     'opcode' => '000403ee',
                     'param1' => 'slot',
                     'size' => 8,
                     'terse' => 1
                   }
        };

File: src/gen3/other/java/src/lib/naigama/Instructions.java
package lib.naigama;

public class Instructions
{
  public static final int INSTR_ANY = 0x000003e4;
  public static final int INSTR_BACKCOMMIT = 0x000403c0;
  public static final int INSTR_CALL = 0x00040382;
  public static final int INSTR_CATCH = 0x00040393;
  public static final int INSTR_CHAR = 0x000403d7;
  public static final int INSTR_CLOSECAPTURE = 0x00040300;
  public static final int INSTR_COMMIT = 0x00040336;
  public static final int INSTR_CONDJUMP = 0x00080321;
  public static final int INSTR_COUNTER = 0x00080356;
  public static final int INSTR_END = 0x000400d8;
  public static final int INSTR_ENDREPLACE = 0x00000399;
  public static final int INSTR_FAIL = 0x0000034b;
  public static final int INSTR_FAILTWICE = 0x00000390;
  public static final int INSTR_JUMP = 0x00040333;
  public static final int INSTR_MASKEDCHAR = 0x00080365;
  public static final int INSTR_NOOP = 0x00000000;
  public static final int INSTR_OPENCAPTURE = 0x0004039c;
  public static final int INSTR_PARTIALCOMMIT = 0x000403b4;
  public static final int INSTR_QUAD = 0x0004037e;
  public static final int INSTR_RANGE = 0x000803bd;
  public static final int INSTR_REPLACE = 0x00080348;
  public static final int INSTR_RET = 0x000003a0;
  public static final int INSTR_SET = 0x002003ca;
  public static final int INSTR_SKIP = 0x00040330;
  public static final int INSTR_SPAN = 0x002003e1;
  public static final int INSTR_TESTANY = 0x00040306;
  public static final int INSTR_TESTCHAR = 0x0008039a;
  public static final int INSTR_TESTQUAD = 0x000803db;
  public static final int INSTR_TESTSET = 0x00240363;
  public static final int INSTR_TRAP = 0xff00ffff;
  public static final int INSTR_VAR = 0x000403ee;


  public static int getSize
    (int opcode)
  {
    return ((opcode >> 16) & 0xff) + 4;
  }

  public static String getString
    (int opcode)
  {
    switch (opcode) {
    case 0x000003e4: return "any";
    case 0x000403c0: return "backcommit";
    case 0x00040382: return "call";
    case 0x00040393: return "catch";
    case 0x000403d7: return "char";
    case 0x00040300: return "closecapture";
    case 0x00040336: return "commit";
    case 0x00080321: return "condjump";
    case 0x00080356: return "counter";
    case 0x000400d8: return "end";
    case 0x00000399: return "endreplace";
    case 0x0000034b: return "fail";
    case 0x00000390: return "failtwice";
    case 0x00040333: return "jump";
    case 0x00080365: return "maskedchar";
    case 0x00000000: return "noop";
    case 0x0004039c: return "opencapture";
    case 0x000403b4: return "partialcommit";
    case 0x0004037e: return "quad";
    case 0x000803bd: return "range";
    case 0x00080348: return "replace";
    case 0x000003a0: return "ret";
    case 0x002003ca: return "set";
    case 0x00040330: return "skip";
    case 0x002003e1: return "span";
    case 0x00040306: return "testany";
    case 0x0008039a: return "testchar";
    case 0x000803db: return "testquad";
    case 0x00240363: return "testset";
    case 0xff00ffff: return "trap";
    case 0x000403ee: return "var";
    default: return "Unknown opcode";
    }
  }
}
